monai/bundle/scripts.py

```diff
@@ -1539,3 +1539,115 @@ def init_bundle(
+
+
+class BundleManager:
+    """
+    The `BundleManager` class facilitates the automatic downloading and instantiation of bundles.
+    It allows users to retrieve bundle properties and meta information.
+
+    Typical usage examples:
+
+    .. code-block:: python
+
+        from monai.bundle import BundleManager
+
+        # Create a BundleManager instance for the 'spleen_ct_segmentation' bundle
+        bundle = BundleManager("spleen_ct_segmentation")
+
+        # Get properties defined in `TrainProperties` or `InferProperties`
+        train_preprocessing = bundle.get("train_preprocessing")
+        print(train_preprocessing)
+
+        # Also support to retrieve meta information from the "metadata.json" file
+        version = bundle.get(meta="version")
+        print(version)
+
+    Args:
+        bundle_name: the name of the bundle. Defaults to None.
+        config_path: file path of the bundle. If a list of file paths is provided,
+            their contents will be merged. Defaults to None.
+        bundle_dir: the target directory to store downloaded bundle.
+            Defaults to the 'bundle' subfolder under `torch.hub.get_dir()`.
+        target directory to store the downloaded data.
+            Default is `bundle` subfolder under `torch.hub.get_dir()`.
+        configs: The name of the config file(s), supporting multiple names.
+            Defaults to "train".
+        kwargs: Additional arguments for download or workflow class instantiation.
+    """
+
+    def __init__(
+        self,
+        bundle_name: str | None = None,
+        config_path: str | Sequence[str] | None = None,
+        bundle_dir: PathLike | None = None,
+        configs: str | Sequence[str] = "train",
+        **kwargs: Any,
+    ) -> None:
+        if bundle_name is None and config_path is None:
+            raise ValueError("Must specify bundle_name or config_path.")
+        configs = ensure_tuple(configs)
+        if "train" in configs:
+            workflow = "train"
+        else:
+            workflow = "infer"
+
+        if config_path is not None:
+            _config_path = Path(ensure_tuple(config_path)[0])
+            if _config_path.is_file():
+                config_file = config_path
+                config_root_path = _config_path.parent
+            else:
+                raise FileNotFoundError(f"Cannot find the config file: {config_path}.")
+        else:
+            download_args = {
+                "version": kwargs.pop("version", None),
+                "args_file": kwargs.pop("args_file", None),
+                "source": kwargs.pop("source", download_source),
+            }
+            download(bundle_name, bundle_dir=bundle_dir, **download_args)
+            bundle_dir = _process_bundle_dir(bundle_dir)
+            config_root_path = bundle_dir / bundle_name / "configs"  # type: ignore
+            if len(configs) > 0:
+                config_file = [str(config_root_path / f"{_config}.json") for _config in configs]
```

Comment:
here should be flexible, the postfix could be `.json`, `.yaml`, `.yml`

Reply:
Thanks, have updated in this commit https://github.com/Project-MONAI/MONAI/pull/6835/commits/8fb5076713f5c4a7be6124c037f563420b8fb9c2.